pkg/serializer/serializer.go

```diff
@@ -329,6 +330,15 @@ func (s *Serializer) SendIterableSeries(series marshaler.IterableMarshaler) erro
 	return s.Forwarder.SubmitSeries(seriesPayloads, extraHeaders)
 }
 
+// IsIterableSeriesSupported returns whether `SendIterableSeries` is supported.
+// Should be removed when `serializePayloadJSON` (useV1API && !s.enableJSONStream) will be removed
+func (s *Serializer) IsIterableSeriesSupported() bool {
+	useV1API := !config.Datadog.GetBool("use_v2_api.series")
+
+	// Supported except for serializePayloadJSON
+	return !(useV1API && !s.enableJSONStream)
```

Comment:
```suggestion
	useV2API := config.Datadog.GetBool("use_v2_api.series")

	// Supported except for serializePayloadJSON
	return useV2API || s.enableJSONStream
```

Reply:
+1, made the same suggestion lol

Reply:
I usually don't use negation like that but here I would like to emphasize the function returns `false` for the case https://github.com/DataDog/datadog-agent/blob/1ae537d530c79333d66e3610168bc1da335cdad3/pkg/serializer/serializer.go#L357
I think `useV2API || s.enableJSONStream` is less obvious that it excludes the case https://github.com/DataDog/datadog-agent/blob/1ae537d530c79333d66e3610168bc1da335cdad3/pkg/serializer/serializer.go#L357.

If you have a strong opinion on it I can still update the code.

Reply:
I don't have a very strong opinion, but I _think_ I still prefer the alternative. If you want to make something very obvious, maybe just add a comment to make explicit whatever context you think is necessary?

But if you'd like to let the code speak for itself, feel free to leave as-is.